assembly/plainyearmonth.ts

```diff
@@ -265,13 +267,21 @@ export class PlainYearMonth {
 
   static compare(a: PlainYearMonth, b: PlainYearMonth): i32 {
     if (a === b) return 0;
-    return compareTemporalDate(
-      a.year,
+    return compare(
+      [a.year,
       a.month,
-      a.referenceISODay,
-      b.year,
+      a.referenceISODay],
+      [b.year,
       b.month,
-      b.referenceISODay
+      b.referenceISODay]
```

Comment:
Using arrays here is unnecessary. I will only create additional gc allocs and slowdowns